multi-agent/lib/mas/core/hitl/ports.py

```
@@ -0,0 +1,122 @@
+    The domain runner depends on this abstraction; the adapter provides
+    the concrete wiring (which gate implementation, which decorator,
+    where to read overrides from).
+
+    Returns ``(None, None)`` when HITL is not available for the given
+    channel (e.g. channel does not support input).
+    """
+
+    @abstractmethod
+    def create(
+        self,
+        channel: Any,
```

Ruff (0.15.20)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `channel`

(ANN401)